### ISSUE TITLE
feature request: add an option to exclude channelgroups and/or recieving device from predictive tuning.

### ISSUE THREAD

**Issue by Leatherface75** [kodi-pvr/pvr.hts#409]
This option would be good if you have a few different channel sources and only want predictive tuning on 1 of the recieving device like HDhomerun with 4 tuners but not for for others.

**Comment by ksooo** [comment]
Not sure I understand what you mean. Could you please give some more details.

**Comment by Leatherface75** [comment]
With HDHomeRun i have 4 tuners and there it's no problem but if you have other sources added in TVheadend it will try to tune a few predictive tuning channels even if it only have 1 tuner.
I only want predictive tuner channels on my HDHomerun tuner and the rest it will work as normal with no predictive tuner channels. For now i have it disabled because of that reason.

**Comment by ksooo** [comment]
I'm afraid tvheadend does not expose how many tuners are available for a given channel. Without this info being available the addon cannot implement what you suggested.

**Comment by Leatherface75** [comment]
Well but we have information about groups so atleast that would be possible.
Not perfect but better than nothing.
Another solution would be support for multiple servers with separate settings for each.
Then i could connect with 2 different users with different channellists.

**Comment by Jalle19** [comment]
You can effectively neuter predictive tuning for a specific network by setting max input streams to 1.

**Comment by Leatherface75** [comment]
Yes i know and I have already done so but it tries to open a lot of channels so zapping is slower instead of faster.
Thats the problem. 
The only solution is to have it disabled for now.

**Comment by Jalle19** [comment]
I don't see why it would be slower? Predictive tuning happens in the background.

**Comment by Leatherface75** [comment]
Because it tries to open the channel and fails to to that and you will have a black screen until it has released all other opened streams like previous one.
The result is that it takes long time while zapping and the only way to speed it up is to disable it.

**Comment by malvinas2** [comment]
+1

For example, IPTV-streams optionally should be excluded from predictive tuning: 
Within tvheadend I configured several streams which rely on streamlink to work properly, a typical call would be: 
```
#EXTINF:-1 tvg-name="Todo Noticias" tvg-id="c5n.com.ar" group-title="youtube", Todo Noticias
pipe:///opt/streamlink.sh http://www.youtube.com/watch?v=-1xif50QMr4
```
with streamlink.sh looking like: 
```
/usr/local/bin/streamlink --hds-segment-threads "3" --stdout "$1" "477p,720p,best"  | ffmpeg -loglevel fatal -i pipe:0 -vcodec copy -acodec ac3 -mpegts_service_type advanced_codec_digital_hdtv -f mpegts pipe:1
```
The problem is that, when a channel is going to be opened, a second instance of streamlink is called nearly simultaneously to get the subsequent channel. For some reasons I don't know, the whole call gets "swallowed" / "stalled", with the result that none of the stream gets opened at all. 
I don't know if this is an error of Kodi, hts, tvheadend or streamlink or because of the feeble single-board computer. 
(also irgendwie wird der ganze Aufruf "verschluckt")

EDIT: Probably a better example of channels taken some time to get opened would be something like `http://www.atresplayer.com/directos/neox/`, where streamlink first has to extract the 'real' address and evaluate the available qualities of the stream, but this spanish channel is geo-blocked. (the whole procedure of streamlink to open a single channel can take up to 10 seconds, running on an Odroid C2 with Ubuntu 16.04.
